event/src/main/java/jetbrains/jetpad/event/TextContentHelper.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright 2012-2016 JetBrains s.r.o
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package jetbrains.jetpad.event;
+
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+
+public class TextContentHelper {
+  private static final char CR = '\r';
+  private static final char LF = '\n';
```

Comment:
That's carriage return, sometimes used on Win systems and some network protocols. We won't use this as separator but it would be nice to recognize it as part of newline sequence.